vslib/src/sai_vs_interfacequery.cpp

```diff
@@ -92,6 +93,13 @@ sai_status_t sai_api_initialize(
         return SAI_STATUS_FAILURE;
     }
 
+    const char *use_tap_dev = service_method_table->profile_get_value(0, SAI_KEY_VS_HOSTIF_USE_TAP_DEVICE);
```

Comment:
>use_tap_dev [](start = 16, length = 11)

The validity or lifetime of returned string.c_str() is problematic. Suggest capture it in a string, or make profile_get_value() return a string directly.

Reply:
profile_get_value is sai api header defined, it needs to reurn char* since its ansi c compiant, i will change to string